File: src/main/java/carelog/carelog/journal/app/JournalService.java
package carelog.carelog.journal.app;

import carelog.carelog.auth.app.CustomUserDetails;
import carelog.carelog.journal.web.dto.JournalCreateRequest;
import carelog.carelog.journal.web.dto.JournalResponse;

import java.util.List;
import java.util.UUID;

public interface JournalService {

    JournalResponse createJournal(UUID relationPublicId, JournalCreateRequest request, CustomUserDetails userDetails);
    JournalResponse updateJournal(UUID relationPublicId, UUID journalPublicId, JournalCreateRequest request, CustomUserDetails userDetails);
    List<JournalResponse> findAllJournals(UUID relationPublicId, CustomUserDetails userDetails);
    JournalResponse findJournal(UUID relationPublicId, UUID journalPublicId, CustomUserDetails userDetails);
    List<JournalResponse> findJournalHistory(UUID relationPublicId, UUID journalPublicId, CustomUserDetails userDetails);
}

File: src/main/kotlin/carelog/carelog/journal/app/JournalService.kt
package carelog.carelog.journal.app

import carelog.carelog.auth.app.CustomUserDetails
import carelog.carelog.journal.web.dto.JournalCreateRequest
import carelog.carelog.journal.web.dto.JournalResponse
import java.util.UUID

interface JournalService {
    fun createJournal(relationPublicId: UUID, request: JournalCreateRequest, userDetails: CustomUserDetails): JournalResponse
    fun updateJournal(relationPublicId: UUID, journalPublicId: UUID, request: JournalCreateRequest, userDetails: CustomUserDetails): JournalResponse
    fun findAllJournals(relationPublicId: UUID, userDetails: CustomUserDetails): List<JournalResponse>
    fun findJournal(relationPublicId: UUID, journalPublicId: UUID, userDetails: CustomUserDetails): JournalResponse
    fun findJournalHistory(relationPublicId: UUID, journalPublicId: UUID, userDetails: CustomUserDetails): List<JournalResponse>
}
